### DISCUSSION TITLE
How the heck do you use Github?

### DISCUSSION THREAD

**Discussion by Lincon1234** [community/community#21691] [New to GitHub]
<p>Hello everyone,</p>
<p>So I’ve been using freecodecamp and talking to developer friends and they say github is of paramount importance to use if you want to land a developer job someday. I have been through the hello world tutorial and making repositories, but when I see projects to work on, how do I work on them? Where do I code/access the existing code? This is probably a serious noob question so I appreciate the response. This stuff certainly isn’t intuitive for me.</p>

**Comment by Am-per-Sand** [comment, marked as answer]
<p>I agree – this is not at all intuitive to a noob and full of hermetic slang.</p>
<p>Basically it is for collaborative edition of files containing the code. Further you are able to review changes, display changes history, discuss solutions etc. You can use this not only for code, but for texts, translations or whatever what you want to track changes of. As soon as you create the repository with files you can use it for all the above.</p>
<p>I use only <a href="http://github.com" rel="nofollow noopener">github.com</a> webpages for installing  git locally is beyond my level.</p>

**Comment by wei** [comment]
<p>Once you identify the project to work on, you can fork it and clone the repo locally to start working on it. <a href="https://code.tutsplus.com/tutorials/how-to-collaborate-on-github--net-34267" rel="nofollow noopener">Here is a good article</a> to learn more about collaboration on Github.</p>

**Comment by oo7Robin** [comment]
<p>I am in the same boat as you! Even after the starting tutorial I felt lost, I recommend watching some youtube video because having someone voice over and guide you through the website is extremely helpful. It will save you a lot of time wandering trying to figure things out on your own. Hope this helps! </p>
<p>Helpful video I found for you: <a href="https://www.youtube.com/watch?v=0fKg7e37bQE" rel="nofollow noopener">https://www.youtube.com/watch?v=0fKg7e37bQE</a></p>
<ul>
<li>Robin.</li>
</ul>

**Comment by mpboom** [comment]
<p>As others suggested, the main workflow is:</p>
<ol>
<li>Find something to work on</li>
<li>Discuss with others in Issues (optionally)</li>
<li>Fork the repo</li>
<li>Clone the repo</li>
<li>Locally create a new branch</li>
<li>Push the local branch to your fork</li>
<li>Make a PR</li>
<li>Discuss the PR, improve the PR and merge</li>
</ol>
<p>As always, if you have any specific questions about Git or GitHub, you are more than welcome to open up a topic over here!</p>
